Source/Csla.AspNetCore/Blazor/ApplicationContextManagerBlazor.cs

```diff
@@ -0,0 +1,166 @@
+using Csla.Core;
+using Microsoft.AspNetCore.Components.Authorization;
+using System;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace Csla.AspNetCore.Blazor
+{
+  /// <summary>
+  /// Application context manager that uses HttpContextAccessor when 
+  /// resolving HttpContext to store context values.
+  /// </summary>
+  public class ApplicationContextManagerBlazor : IContextManager, IDisposable
+  {
+    private ContextDictionary LocalContext { get; set; }
+    private ContextDictionary ClientContext { get; set; }
+    private IPrincipal CurrentPrincipal { get; set; }
+    private readonly ClaimsPrincipal UnauthenticatedPrincipal = new();
+    private bool disposedValue;
+
+    /// <summary>
+    /// Gets the current HttpContext instance.
+    /// </summary>
+    protected AuthenticationStateProvider AuthenticationStateProvider { get; private set; }
+
+    /// <summary>
+    /// Gets or sets a reference to the current ApplicationContext.
+    /// </summary>
+    public ApplicationContext ApplicationContext { get; set; }
+
+    /// <summary>
+    /// Creates an instance of the object, initializing it
+    /// with the required IServiceProvider.
+    /// </summary>
+    /// <param name="authenticationStateProvider">AuthenticationStateProvider service</param>
+    public ApplicationContextManagerBlazor(AuthenticationStateProvider authenticationStateProvider)
+    {
+      AuthenticationStateProvider = authenticationStateProvider;
+      CurrentPrincipal = UnauthenticatedPrincipal;
+      AuthenticationStateProvider.AuthenticationStateChanged += AuthenticationStateProvider_AuthenticationStateChanged;
+      InitializeUser();
+    }
+
+    private void InitializeUser()
+    {
+      AuthenticationStateProvider_AuthenticationStateChanged(AuthenticationStateProvider.GetAuthenticationStateAsync());
+    }
+
+    private void AuthenticationStateProvider_AuthenticationStateChanged(Task<AuthenticationState> task)
+    {
+      task.ContinueWith((t) =>
+      {
+        if (task.IsCompletedSuccessfully && task.Result != null)
+          CurrentPrincipal = task.Result.User;
+        else
+          CurrentPrincipal = UnauthenticatedPrincipal;
+      });
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether this
+    /// context manager is valid for use in
+    /// the current environment.
+    /// </summary>
+    public bool IsValid
+    {
+      get { return true; }
+    }
+
+    /// <summary>
+    /// Gets the current principal.
+    /// </summary>
+    public IPrincipal GetUser()
+    {
+      return CurrentPrincipal;
+    }
+
+    /// <summary>
+    /// Attempts to set the current principal on the registered
+    /// IHostEnvironmentAuthenticationStateProvider service.
+    /// </summary>
+    /// <param name="principal">Principal object.</param>
+    public virtual void SetUser(IPrincipal principal)
+    {
+      CurrentPrincipal = principal;
```

Comment:
Could be a dumb suggestion...but I found that your LocalProxy was calling this so often as I jumped around on the server side of the dataportal (many child objects) and so I put a check in here to see if the CurrentPrincipal == principal then skip the whole SetAuthenticationState which seemed to me to just cause needless consumption of compute resources.